### PR TITLE
feat: Add warning about NaN invariant and -0.0 for direct comparisons

### PR DESCRIPTION
**Pull request by MaxGraey**
Produce warning like:
```
WARNING AS907: Comparision with NaN is invariant. You need to use isNaN(x).
      :
    2 │ assert(x != NaN);
      │        ~~~~~~~~
      └─ in index.ts(2,8)
```
and
```
WARNING AS907: Comparision with NaN is invariant. You need to use isNaN(x).
      :
    3 │ assert(x == NaN);
      │        ~~~~~~~~
      └─ in index.ts(3,8)
```
also
```
WARNING AS908: Comparision with -0.0 is sign insensitive. Perhaps you want to use Object.is(x, -0.0).
      :
    4 │ assert(x != -0.0);
      │        ~~~~~~~~~
      └─ in index.ts(4,8)
```
- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/diagnosticMessages.json [hunk 1]
```diff
@@ -56,6 +56,8 @@
   "Indexed access may involve bounds checking.": 904,
   "Explicitly returning constructor drops 'this' allocation.": 905,
   "Unnecessary definite assignment.": 906,
+  "Comparision with NaN is invariant. You need to use isNaN(x).": 907,
```
Comment (dcodeIO) [inline src/diagnosticMessages.json:59]: Perhaps
```suggestion
  "Comparison with NaN is always false. Use isNaN(x) instead.": 907,
```

Comment (MaxGraey) [reply src/diagnosticMessages.json:59]: > "Comparison with NaN is always false

That's not precise. `x == NaN` always false. `x != NaN` always true

File: src/diagnosticMessages.json [hunk 2]
```diff
@@ -56,6 +56,8 @@
...
+  "Comparision with -0.0 is sign insensitive. Perhaps you want to use Object.is(x, -0.0).": 908,
```
Comment (dcodeIO) [inline src/diagnosticMessages.json:60]:
```suggestion
  "Comparison with -0.0 is sign insensitive. Use Object.is(x, -0.0) if the sign matters.": 908,
```

File: src/diagnosticMessages.json (hunk 1 above)
Comment (dcodeIO) [reply src/diagnosticMessages.json:59]:
```suggestion
  "NaN does not compare equal to any other value including itself. Use isNaN(x) instead.": 907,
```

**Pull request merged**